src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -65,7 +65,7 @@ public function __toString()
      *
      * @param string|null $key The name of the headers to return or null to get them all
      *
-     * @return array<string, array<int, string|null>>|array<int, string|null>
+     * @return $key is null ? array<string, array<int, string|null>> : array<int, string|null>
```

Comment:
```suggestion
     * @return ($key is null ? array<string, array<int, string|null>> : array<int, string|null>)
```
See https://phpstan.org/r/18e18b32-1cc3-4647-875b-c756ad1ad4d7

Also, to keep IDE support I would recommend
```suggestion
     * @return array<string, array<int, string|null>>|array<int, string|null>
     * @phpstan-return ($key is null ? array<string, array<int, string|null>> : array<int, string|null>)
```
(or psalm-return if preferred)

Reply:
Applied the first fix, thanks.

The second I would agree but I think it's again symfony policy so leaving it as is.